instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/jdbc/OpenTelemetryJdbcDriverAutoConfiguration.java

```diff
@@ -1,47 +0,0 @@
-/*
- * Copyright The OpenTelemetry Authors
- * SPDX-License-Identifier: Apache-2.0
- */
-
-package io.opentelemetry.instrumentation.spring.autoconfigure.instrumentation.jdbc;
-
-import io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver;
-import io.opentelemetry.instrumentation.spring.autoconfigure.OpenTelemetryInjector;
-import io.opentelemetry.instrumentation.spring.autoconfigure.internal.SdkEnabled;
-import org.springframework.beans.factory.config.BeanDefinition;
-import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
-import org.springframework.boot.autoconfigure.AutoConfigureAfter;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Conditional;
-import org.springframework.context.annotation.Configuration;
-
-@ConditionalOnClass(OpenTelemetryDriver.class)
-@ConditionalOnProperty(
-    name = "spring.datasource.driver-class-name",
-    havingValue = "io.opentelemetry.instrumentation.jdbc.OpenTelemetryDriver")
-@Conditional(SdkEnabled.class)
-@Configuration(proxyBeanMethods = false)
-@AutoConfigureAfter(
-    name = "org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration")
-@ConditionalOnBean(name = "org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration")
-public class OpenTelemetryJdbcDriverAutoConfiguration {
```

Comment:
Remove in favor of the new implementation that does not require to set the OpenTelemetry driver and update the datasource url with `otel:`.

The new implementation does not require any user configuration to benefit from JDBC instrumentation.